pam/integration-tests/vhs-helpers_test.go

```diff
@@ -322,3 +323,36 @@ func evaluateTapeVariables(t *testing.T, tapeString string, td tapeData) string
 
 	return tapeString
 }
+
+func requireRunnerResultForUser(t *testing.T, sessionMode authd.SessionMode, user, goldenContent string) {
+	t.Helper()
+
+	var msgFormat string
+	switch sessionMode {
+	case authd.SessionMode_AUTH:
+		msgFormat = pam_test.RunnerResultActionAuthenticateFormat
+	case authd.SessionMode_PASSWD:
+		msgFormat = pam_test.RunnerResultActionChangeAuthTokFormat
+	default:
+		t.Errorf("Unsupported mode %s", sessionMode)
+	}
+
+	sessionMsg := fmt.Sprintf(msgFormat, user)
+	if user == "" {
+		sessionMsg = strings.ReplaceAll(msgFormat, "%q", "")
+	}
```

Comment:
I don't like that we reimplement assembling of that message here. This will break if anything changes in https://github.com/3v1n0/authd/blob/572504e1e0c10d14d2e2afb89018023fb5c3c04f/pam/tools/pam-runner/pam-runner.go#L118-L133. Can we extract a function like this:

```
func PamResultMessage(mode string, user string) string {
	switch mode {
	case "login":
		return fmt.Sprintf("PAM ChangeAuthTok() for user %q", user)
	case "passwd":
		return fmt.Sprintf("PAM Authenticate() for user %q", user)
	default:
		panic("Invalid PAM operation: " + mode)
	}
}
```

and use it in both places?

Reply:
I had done that exactly to avoid breakage, but ok we can instead have a function instead I think, under the `pam_test` namespace.

Reply:
> I had done that exactly to avoid breakage

What do you mean? That you reused the `pam_test.RunnerResultActionChangeAuthTokFormat` and `pam_test.RunnerResultActionAuthenticateFormat` constants instead of copying the string literal? That's a good first step, but still quite brittle IMO.

Reply:
Ah, I guess you mean 742b9c68d60ece73f509a445187aa6ff8b7c19cf where you extracted the constants

Reply:
Yeah, let me reconsider your approach... Will handle ti soon :)